src/Compilers/Core/Desktop/CommandLine/CommonCompiler.cs

```diff
@@ -212,16 +214,8 @@ internal bool PrintErrors(IEnumerable<Diagnostic> diagnostics, TextWriter consol
                     continue;
                 }
 
-                // Catch exceptions from diagnostic formatter as diagnostic descriptors for analyzer diagnostics can throw an exception while formatting diagnostic message.
```

Comment:
yes, John made the LocalizableString intrinsically exception safe, so no API on diagnostic can now throw.